src/os.cpp

```diff
@@ -1456,7 +1456,10 @@ static void init_rand() {
     memcpy(&seed, bytes, sizeof(unsigned));
     srand(seed);
 #elif defined(ZIG_OS_LINUX)
-    srand(*((unsigned*)getauxval(AT_RANDOM)));
+    unsigned *ptr_random = (unsigned*)getauxval(AT_RANDOM);
+    int seed;
+    memcpy(&seed, ptr_random, sizeof(unsigned));
```

Comment:
`s/sizeof(unsigned)/sizeof(seed)`, otherwise you're stomping the stack on 64bit arches.
Also `srand` accepts unsigned ints.